### DISCUSSION TITLE
[Suggest] Provide separate @types file

### DISCUSSION THREAD

**Discussion by KAWATAAAAAA** [element-plus/element-plus#1473] [Ideas]
## Compositon Api
```js
import { ElLoading,ILoadingInstance } from "element-plus"

export const useLoadingHook = () => {
    const loadingInstance:ILoadingInstance = null

    function startLoading(text?:string){
        ElLoading.service({
            lock: true,
            text: text || '拼命加载中',
            spinner: 'el-icon-loading',
            background: 'rgba(0, 0, 0, 0.7)'
        });
    }

    function closeLoading(){
        ( loadingInstance as ILoadingInstance).close()
    }


    return {
        startLoading,
        closeLoading
    }
}
```
## idea told me ：
> 
>Module '"../../node_modules/element-plus/lib"' has no exported member 'ILoadingInstance'. Did you mean to use 'import ILoadingInstance from "../../node_modules/element-plus/lib"' instead?
>

what does the "../../node_modules" is ？，I think it well take effect，but it's too long .....

I see on the vue/vite project package.json file：
like vite ："types": "link:./packages/vite/types",
like vue ："types": "dist/vue.d.ts",

maybe the best way is this？

thanks! 如果没看懂我的意思，那就是我英语不好😅

**Comment by KAWATAAAAAA** [comment]
### Actually  
> import ILoadingInstance from "../../node_modules/element-plus/lib" ❌ It doesn't work
> import { ILoadingInstance } from "../../node_modules/element-plus/lib/el-loading/src/loading.type" ✅

**Reply by jw-foss** [reply]
Typing issue has been the pain for a long period of time. But the team is quite overloaded with issues, features. We can't make any promises right now for any period of time that we could deliver the typing feature to our users. But we will try our best to do that ;)
